File: src/main/java/org/apache/prepbuddy/datacleansers/imputation/Probability.java
package org.apache.prepbuddy.datacleansers.imputation;

import org.apache.prepbuddy.exceptions.ApplicationException;
import org.apache.prepbuddy.exceptions.ErrorMessages;

import java.io.Serializable;
import java.text.DecimalFormat;

public class Probability implements Serializable {
    private double probability;

    public Probability(double probability) {
        if (probability < 0 || probability > 1)
            throw new ApplicationException(ErrorMessages.PROBABILITY_IS_NOT_IN_RANGE);
        probability = Double.parseDouble(new DecimalFormat("##.####").format(probability));
        this.probability = probability;

    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Probability that = (Probability) o;

        return Double.compare(that.probability, probability) == 0;

    }


    public Double value() {
        return probability;
    }
}

File: src/main/scala/org/apache/datacommons/prepbuddy/utils/Probability.scala
package org.apache.datacommons.prepbuddy.utils

import org.apache.datacommons.prepbuddy.exceptions.{ApplicationException, ErrorMessages}

class Probability(probabilityValue: Double) extends Serializable {
    if (probabilityValue > 1 || probabilityValue < 0) {
        throw new ApplicationException(ErrorMessages.PROBABILITY_IS_NOT_IN_RANGE)
    }
    def divide(other: Probability): Probability = {
        new Probability(this.probabilityValue / other.doubleValue)
    }

    def multiply(other: Probability): Probability = {
        new Probability(this.probabilityValue * other.doubleValue)
    }

    def doubleValue: Double = probabilityValue
}
